mmit/tests/test_solver.py

```diff
@@ -299,20 +295,22 @@ def test_real_data_4(self):
-        target_upper = np.array([-inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, 0.09506, 0.100839, 0.08891, 0.102143, inf, 0.10514, 0.088991, 0.129059, 0.094515, 0.083031, inf, 0.081022, 0.091533, 0.10649, 0.088369, 0.096142, 0.103672, 0.076665, 0.10408, 0.083655, 0.805308, 0.781073, 0.789102, 0.798496, 0.782113, 0.794679, 0.79917, 0.796532, 0.783136, 0.7851, inf, 0.783219, 0.790679, 0.796257, 0.802003, 0.788213, 0.791343, 0.798649, 0.791274, 0.801812, 1.195363, 1.200608, 1.202095, 1.201477, inf, inf, 1.207487, 1.197906, 1.202613, 1.194013, 1.202462, 1.205496, 1.216635, 1.20226, 1.203679, 1.201785, 1.198553, 1.204157, inf, 1.204639, 1.487494, 1.494513, 1.473028, 1.477199, 1.480064, 1.498114, 1.48911, 1.502461, 1.480309, 1.484982, 1.485259, 1.479859, 1.49189, 1.491625, 1.491892, 1.471031, 1.492252, 1.480454, 1.477445, 1.728685, 1.481343, 1.70353, inf, 1.713546, 1.722064, 1.719361, 1.676259, inf, 1.715813, 1.709235, inf, 1.684199, 1.729788, 1.723125, 1.691868, 1.713605, 1.709298, 1.70427, 1.713183, 1.708068, inf, 1.917625, inf, 1.876062, 1.911192, 1.894552, 1.902184, 1.898814, 1.88457, 1.888766, 1.881155, inf, 1.896417, 1.888765, 1.882561, 1.891158, 1.90534, 1.903245, 1.874546, 1.898803, 2.063508, 2.052022, 2.03937, 2.044155, 2.052685, 2.049617, inf, 2.029818, 2.041099, inf, 2.045597, 2.050845, 2.03713, 2.043467, 2.029973, 2.046673, 2.047936, 2.043105, 2.048092, 2.054786, 2.16948, 2.172143, 2.171181, 2.197085, 2.165541, 2.192424, 2.191335, 2.193998, 2.174704, 2.172875, 2.180607, 2.172124, 2.176899, 2.179441, 2.166317, 2.161891, 2.191185, inf, 2.188154, 2.190848, 2.303124, 2.312501, 2.291121, 2.310343, inf, 2.301484, inf, 2.309122, 2.309144, 2.303975, 2.295898, 2.299242, 2.295398, 2.299057, 2.313572, 2.288616, 2.313304, 2.30247, 2.310038, 2.306447])
+        margin = 0.5  # Should not affect the solution
+        values = np.random.rand(1000)
+        moves, preds, costs = solver.compute_optimal_costs(values, values, margin, 1)  # squared hinge
+        np.testing.assert_almost_equal(preds[-1], np.mean(values))
```

Comment:
you could also use cumsum and test all of the values (instead of just the last one)